app/controllers/application_controller.rb

```diff
@@ -12,28 +12,36 @@ def index
   end
 
   def new
-    @collection = model_class.new
+    if class_name == "List"
```

Comment:
I'd move "List" into some constant, and also I'd suggested to move this logic into some builder service